### PR TITLE
fix: [CI-3966]: pool name

### PR DESCRIPTION
**Pull request by mtran7**
##### Screenshots:

Changed from Pool ID to Pool Name
image

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

**Comment by mtran7** [comment]
retrigger titlecheck

File: src/modules/70-pipeline/components/PipelineStudio/StepUtil.ts [hunk 1]
```diff
@@ -271,6 +272,38 @@ export const validateStage = ({
         set(errors, 'spec.infrastructure.infrastructureDefinition.spec', errorsResponse)
       }
     }
+    // CI validation
+    if (
+      viewType !== StepViewType.InputSet &&
+      isEmpty((stageConfig?.infrastructure as K8sDirectInfraYaml)?.spec?.connectorRef) &&
+      (stageConfig?.infrastructure as K8sDirectInfraYaml)?.type === 'KubernetesDirect' &&
```
Comment (mtran7) [inline src/modules/70-pipeline/components/PipelineStudio/StepUtil.ts:279]: Cannot reference ci constants but shouldn't duplicate own so keeping string for now

File: src/modules/70-pipeline/components/PipelineInputSetForm/StageInputSetForm.tsx [hunk 2]
```diff
@@ -615,6 +615,32 @@ export function StageInputSetFormInternal({
     [expressions]
   )
 
+  const renderVMInfra = (): JSX.Element | null => {
```
Comment (vardanbansal-harness) [inline src/modules/70-pipeline/components/PipelineInputSetForm/StageInputSetForm.tsx:618]: maybe wrap in a `useCallback`?

**Review by vardanbansal-harness — DISMISSED** [review]

File: src/modules/70-pipeline/components/PipelineInputSetForm/StageInputSetForm.tsx [hunk 3]
```diff
@@ -624,6 +624,105 @@ export function StageInputSetFormInternal({
...
+                gitScope={gitScope}
+                disabled={readonly}
+                type={Connectors.DOCKER}
+              />
+            </Container>
+          )
+        ) : null}
+      </>
+    )
+  }, [expressions])
```
Comment (vardanbansal-harness) [inline src/modules/70-pipeline/components/PipelineInputSetForm/StageInputSetForm.tsx:724]: One thing I noticed in few of my last PR was I was not including all the dependencies that I should have calling this `useCallback` not changing as per those missing dependencies. For e.g. can you verify if this returns the right behaviour when `readonly` is `true` vs `false` ?

**Review by vardanbansal-harness — DISMISSED** [review]

**Review by Tamaraius — DISMISSED** [review]

File: src/modules/70-pipeline/components/PipelineInputSetForm/StageInputSetForm.tsx (hunk 3 above)
Comment (mtran7) [reply src/modules/70-pipeline/components/PipelineInputSetForm/StageInputSetForm.tsx:724]: Yup, verified passing readonly true makes the fields disabled

**Pull request merged**